crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -104,6 +181,11 @@ impl DynamicStruct {
         }
     }
 
+    /// Inserts a field named `name` with the typed value `value` into the struct.
+    ///
+    /// If the field already exists, it is overwritten.
+    ///
+    /// The type information of the inserted value is erased.
```

Comment:
This isn't entirely true. Downcasting is still possible and only works for the original type. _Compile-time_ type information is erased.

Reply:
And I'm not sure including a line about that is particularly relevant. Most Rust libs I've seen don't include lines like this every time they accept a generic type that gets boxed internally.